### PR TITLE
Remove setting None to TensorBoardLogger warning

### PR DESCRIPTION
**Pull request by kaushikb11**
## What does this PR do?

Was added in #10777

The warning won't be necessary for majority of users.

And this warning pops up every time a user trains with `ddp_spawn`,` ddp_spawn_sharded` and `tpu_spawn` as TensorBoard logger is the default.

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @borda

### CODE REVIEW

File: pytorch_lightning/plugins/training_type/ddp_spawn.py [hunk 1]
```diff
@@ -425,9 +425,6 @@ def _clean_logger(trainer: "pl.Trainer") -> None:
         loggers = trainer.logger._logger_iterable if isinstance(trainer.logger, LoggerCollection) else [trainer.logger]
         for logger in loggers:
             if isinstance(logger, TensorBoardLogger) and logger._experiment is not None:
-                rank_zero_warn(
```
Comment (rohitgr7) [inline pytorch_lightning/plugins/training_type/ddp_spawn.py:428]: I think if we are setting it to None below then we are creating a new experiment after log_hyperparams is called in the same trainer.fit call

File: CHANGELOG.md [hunk 2]
```diff
@@ -182,6 +182,9 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 - Removed `DeepSpeedPlugin.{precision,amp_type,amp_level}` properties ([#10657](https://github.com/PyTorchLightning/pytorch-lightning/pull/10657))
 
 
+- Removed setting None to `TensorBoardLogger` warning in `DDPSpawn` plugin ([#10805](https://github.com/PyTorchLightning/pytorch-lightning/pull/10805))
```
Comment (carmocca) [inline CHANGELOG.md:185]: No need for this update. This is unreleased.

**Review by carmocca — APPROVED** [review]

File: pytorch_lightning/plugins/training_type/ddp_spawn.py (hunk 1 above)
Comment (ananthsub) [reply pytorch_lightning/plugins/training_type/ddp_spawn.py:428]: could this be a rank_zero_debug instead of a warning?

**Review by awaelchli — APPROVED** [review]
the fix didn't work in the end, so it's fine to remove it again.

File: CHANGELOG.md [hunk 3]
```diff
@@ -182,6 +182,8 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 - Removed `DeepSpeedPlugin.{precision,amp_type,amp_level}` properties ([#10657](https://github.com/PyTorchLightning/pytorch-lightning/pull/10657))
 
 
+
+
```
Comment (kaushikb11) [inline CHANGELOG.md:186]:
```suggestion
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -182,6 +182,8 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
...
 ### Fixed
```
Comment (kaushikb11) [inline CHANGELOG.md:187]:
```suggestion

### Fixed
```

File: pytorch_lightning/plugins/training_type/ddp_spawn.py (hunk 1 above)
Comment (kaushikb11) [reply pytorch_lightning/plugins/training_type/ddp_spawn.py:428]: The fix didn't work, we could remove it.

**Review by tchaton — APPROVED** [review]
LGTM !

**Review by Borda — APPROVED** [review]

**Comment by rohitgr7** [comment]
is this required now? https://github.com/PyTorchLightning/pytorch-lightning/pull/10838?

**Pull request closed without merging**